goenv/goenv.go

```diff
@@ -67,11 +70,98 @@ func Get(name string) string {
 		return "1"
 	case "TINYGOROOT":
 		return sourceDir()
+	case "WASMOPT":
+		if path := os.Getenv("WASMOPT"); path != "" {
+			err := wasmOptCheckVersion(path)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "cannot use %q as wasm-opt (from WASMOPT environment variable): %s", path, err.Error())
+				os.Exit(1)
+			}
+
+			return path
+		}
+
+		return findWasmOpt()
 	default:
 		return ""
 	}
 }
 
+// Find wasm-opt, or exit with an error.
+func findWasmOpt() string {
```

Comment:
[this comment was intended to be part of the review but I forgot to click the "start review" button]

I think this function can be simplified. In short: the bundled wasm-opt binaries don't need to be checked for a version.

I would suggest the following logic:

```
if $TINYGOROOT/bin/wasm-opt[.exe] exists:
    return this path
if $TINYGOROOT/build/wasm-opt[.exe] exists:
    return this path
look for wasm-opt using LookPath
if not found:
    print or return error
check for version
if version too old:
    return error
return this path
```

This is not a blocker to me.

Reply:
I think we should still be doing a version check regardless for now.

Reply:
This has already caught one issue, so it is still helpful.